assets/src/dashboard/app/views/templatesGallery/index.js

```diff
@@ -0,0 +1,160 @@
+import { ApiContext } from '../../api/apiProvider';
+import { UnitsProvider } from '../../../../edit-story/units';
+import { TransformProvider } from '../../../../edit-story/components/transform';
+import FontProvider from '../../font/fontProvider';
+import usePagePreviewSize from '../../../utils/usePagePreviewSize';
+import {
+  BodyWrapper,
+  PageHeading,
+  NoResults,
+  StoryGridView,
+  ListBarContainer,
+} from '../shared';
+
+const ExploreFiltersContainer = styled.div`
+  padding: 0 20px 20px;
+  border-bottom: ${({ theme }) => theme.subNavigationBar.border};
+  flex-direction: row;
+  display: flex;
+
+  ${DropdownContainer} {
+    margin-right: 15px;
+  }
+`;
+
+function TemplatesGallery() {
+  const [typeaheadValue, setTypeaheadValue] = useState('');
+  const [viewStyle, setViewStyle] = useState(VIEW_STYLE.GRID);
+  const { pageSize } = usePagePreviewSize();
+  const {
+    state: { templates },
+  } = useContext(ApiContext);
+
+  const filteredTemplates = useMemo(() => {
+    return templates.filter((template) => {
+      const lowerTypeaheadValue = typeaheadValue.toLowerCase();
+      return template.title.toLowerCase().includes(lowerTypeaheadValue);
+    });
+  }, [templates, typeaheadValue]);
+
+  const handleViewStyleBarButtonSelected = useCallback(() => {
+    if (viewStyle === VIEW_STYLE.LIST) {
+      setViewStyle(VIEW_STYLE.GRID);
+    } else {
+      setViewStyle(VIEW_STYLE.LIST);
+    }
+  }, [viewStyle]);
+
+  const filteredTemplatesCount = filteredTemplates.length;
+
+  const listBarLabel = sprintf(
+    /* translators: %s: number of stories */
+    _n(
+      '%s total template',
+      '%s total templates',
+      filteredTemplatesCount,
+      'web-stories'
+    ),
+    filteredTemplatesCount
+  );
+
+  const BodyContent = useMemo(() => {
+    if (filteredTemplatesCount > 0) {
+      return (
+        <BodyWrapper>
+          <ListBarContainer>
+            <ListBar
+              label={listBarLabel}
+              layoutStyle={viewStyle}
+              onPress={handleViewStyleBarButtonSelected}
+            />
+          </ListBarContainer>
+          <StoryGridView filteredStories={filteredTemplates} />
+        </BodyWrapper>
+      );
+    }
+
+    return <NoResults typeaheadValue={typeaheadValue} />;
+  }, [
+    filteredTemplates,
+    filteredTemplatesCount,
+    handleViewStyleBarButtonSelected,
+    listBarLabel,
+    typeaheadValue,
+    viewStyle,
+  ]);
+
+  return (
+    <FontProvider>
+      <TransformProvider>
+        <UnitsProvider pageSize={pageSize}>
+          <PageHeading
+            defaultTitle={__('Explore Templates', 'web-stories')}
+            searchPlaceholder={__('Template Stories', 'web-stories')}
+            filteredStories={filteredTemplates}
+            handleTypeaheadChange={setTypeaheadValue}
+            typeaheadValue={typeaheadValue}
+          />
+          <ExploreFiltersContainer>
+            <Dropdown
+              ariaLabel="Category Dropdown"
```

Comment:
nit: ariaLabels should get i18n treatment too

Reply:
Good catch! I'll fix!